include/bluetooth/conn.h

```diff
@@ -672,8 +672,9 @@ struct bt_conn_auth_cb {
 	/** @brief notify that pairing process has failed.
 	 *
 	 * @param conn Connection object.
+	 * @param reason SMP reason for the failure.
 	 */
-	void (*pairing_failed)(struct bt_conn *conn);
+	void (*pairing_failed)(struct bt_conn *conn, u8_t reason);
```

Comment:
While this seems fine at first glance we can only have one instance of bt_conn_auth_cb registered which is basically an authentication 'agent' that may not be on the application side, which is the reason Ive suggested a new callback in the bt_conn_cb which can have multiple instances.